### PR TITLE
Explosion for A04 androids

### PR DESCRIPTION
**Pull request by Whoever-this-guy-is**
Adds an explosion effect for A04s when killed, the explosion damage is same as SILVER T with only +20 damage (no noticeable effect found so it could be the same i guess?)
Has a condition where the explosion doesnt happen if the unit has no mines left in the inventory
Includes detonation on attack command

### CODE REVIEW

**Comment by michaeldgg2** [comment]
Please remove stuff related to WTRN and WTP100 Cargo/Passenger stuff. We can't use built-in traits as we'll need to properly support vehicles inside them:

https://github.com/OpenE2140/OpenE2140/pull/289#issuecomment-1591948070

File: mods/e2140/content/ed/infantry/a04/rules.yaml [hunk 1]
```diff
@@ -22,9 +22,24 @@ ed_infantry_a04:
 		Name: Mines
 		Armaments: WeaponMines
 		Ammo: 5
+		AmmoCondition: Kamikaze
```
Comment (michaeldgg2) [inline mods/e2140/content/ed/infantry/a04/rules.yaml:25]: Probably a better name for condition would be `MinesAvailable` or `HasMines`. Makes it more obvious what it does in `GrantConditionOnAttack` and `KillsSelf` traits.

File: mods/e2140/content/ed/infantry/a04/rules.yaml [hunk 2]
```diff
@@ -22,9 +22,24 @@ ed_infantry_a04:
...
 	Rearmable:
 		RearmActors:
 		AmmoPools: Mines
+	AttackFrontal:
+		Voice: Attack
+	Armament@PRIMARY:
+		Weapon: ed_infantry_a04
+		MuzzlePalette:
+	GrantConditionOnAttack:
+		Condition: triggered
```
Comment (michaeldgg2) [inline mods/e2140/content/ed/infantry/a04/rules.yaml:40]: This is the condition I would call `kamikaze` (I do know, we use `triggered` for Silver T). I'll let @Dzierzan make a decision about this.

**Review by michaeldgg2 — CHANGES_REQUESTED** [review]

**Comment by Whoever-this-guy-is** [comment]
sorry about the WTRN and WTP mess

**Comment by Whoever-this-guy-is** [comment]
ah you know you can close this PR, too chaotic to pull so might as well re-do it from scratch

**Pull request closed without merging**